### PR TITLE
Fix trigger of tests

### PR DESCRIPTION
**Pull request by kamilchodola**
I've used invalid event to distinguish if job should be triggered or not. Fixed

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [X] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/sync-supported-chains.yml [hunk 1]
```diff
@@ -227,7 +227,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Trigger notification action on test repo
-        if: github.event_name == 'push' && (startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/heads/release/'))
+        if: github.event.workflow_run.head_branch == 'master' || contains(github.event.workflow_run.head_branch, 'release/')
```
Comment (LukaszRozmej) [inline .github/workflows/sync-supported-chains.yml:230]: not sure about that `contains`, shouldn't be `startswith`?

Comment (kamilchodola) [reply .github/workflows/sync-supported-chains.yml]: Valid point - but hard to verify how branch is provided with workflow_run so will verify shortly after merge if all is ok.

"push" prfixes with "refs/heads" where workflow_run should not but GH never stops surprising me. That's whY i chosen contains at first place but may hit some edge case with weird branches like "test/release/1.28.0" (which can happen)
